hnet/handler.go

```diff
@@ -183,6 +183,24 @@ func handleStopSpectating(stream *common.IOStream, player *Player) error {
 	return player.StopSpectating()
 }
 
+func handleHasMap(stream *common.IOStream, player *Player) error {
+	request := ReadHasMapRequest(stream)
+
+	if request == nil {
+		return fmt.Errorf("failed to read has map request")
+	}
+
+	player.LogIncomingPacket(CLIENT_SPECTATE_HAS_MAP, request)
+
+	response := &HasMapResponse{
+		UserId: player.Info.Id,
+		HasMap: request.HasMap,
+	}
+	player.Host.SendPacket(SERVER_SPECTATE_HAS_MAP, response)
```

Comment:
We also need to send this to the host's spectators. So `player.Host.Spectators.Broadcast`.

Reply:
Nevermind, I forgot that there were no fellow spectators.. whoops